crates/ntx-builder/src/builder/mod.rs

```diff
@@ -94,6 +101,87 @@ pub struct NetworkTransactionBuilder {
 }
 
 impl NetworkTransactionBuilder {
+    pub async fn serve_new(self) -> anyhow::Result<()> {
+        // TODO: separate out the startup stuff so it can loop and repeat and wait on the network
+        // etc.
+        let store = StoreClient::new(&self.store_url);
+        let block_prod = BlockProducerClient::new(self.block_producer_address);
+
+        let unconsumed = store
+            .get_unconsumed_network_notes()
+            .await
+            .context("failed to fetch unconsumed notes from the store")?;
+        let (chain_tip, _mmr) = store
+            .get_current_blockchain_data(None)
+            .await
+            .context("failed to fetch the chain tip data from the store")?
+            .context("chain tip data was None")?;
+
+        let mut mempool_events = block_prod
+            .subscribe_to_mempool(chain_tip.block_num())
+            .await
+            .context("failed to subscribe to mempool events")?;
+
+        let mut state = crate::state::State::with_committed_notes(unconsumed.into_iter());
+        let mut interval = tokio::time::interval(self.ticker_interval);
+        interval.set_missed_tick_behavior(time::MissedTickBehavior::Skip);
+
+        // Tracks network transaction tasks until they are submitted to the mempool.
+        //
+        // We also map the task ID to the network account so we can mark it as failed if it doesn't
+        // get submitted.
+        let mut inflight = JoinSet::new();
+        let mut inflight_idx = HashMap::new();
+
+        loop {
+            tokio::select! {
+                _next = interval.tick() => {
+                    if inflight.len() > MAX_IN_PROGRESS_TXS {
```

Comment:
The inflight set is limited by `MAX_INPROGRESS_TXS`. In each of the m`MAX_NOTES_PER_TX` can exist. How do we ensure they're not invalidating one another?

Reply:
This follows from two facts:

1. A network note is bound to a single network account. As in, the note specifies a single network account that is allowed to consume it.
2. The `State` itself locks a network account once it has been selected until it has completed.

So each note can only be consumed by a specific account, and no account can have more than one transaction in-progress at once.